### PR TITLE
update CC Monitoring page with Datadog integration steps (WIP)

### PR DESCRIPTION
**Pull request by taroface**
Fixes DOC-1865.

Draft steps for using the Datadog integration with CC Dedicated. This is a WIP and still needs:

- Name and description of readymade Datadog dashboard
- Link to corresponding Datadog doc, which I assume will include the list of metrics as with the self-hosted integration (see: https://www.cockroachlabs.com/docs/stable/datadog.html)
- ~An update to how `Active` integration status is defined on our end. This currently describes the beta model and is insufficient, because the doc doesn't get into how a customer can define a custom monitor to notify that CRDB metrics aren't being exported. I figured this would change for GA, but if not, let me know.~ Updated per https://github.com/cockroachlabs/managed-service/pull/8511

### CODE REVIEW

File: cockroachcloud/monitoring-page.md
Comment (abarganier) [inline cockroachcloud/monitoring-page.md:46]: > `Inactive` indicates that the integration has not been successfully deployed.

Does the wording here indicate a problem? If the customer never tried setting up the integration in the first place, then "has not been **successfully** deployed" feels like it indicates some error/failure occurred, when the reality is that they just never tried deploying it before. 

However, it's indeed also true that if an error occurred during the setup, the `Inactive` status will show as well, where the error is indicated by the "Datadog integration rollout failed. Retry rollout" text. How can we word this to cover both scenarios? 🤔

Comment (taroface) [reply cockroachcloud/monitoring-page.md:46]: Good point - how about:

"Setup has either not been attempted or has encountered an error."

**Comment by thtruo** [comment]
Heads up @taroface Alex and I hopped off a call with the Datadog folks, and they recommended/suggested we provide a screenshot in docs to show what it looks like to do the integration setup. The flow is straightforward and likely can be represented with 3 screenshots - but in the interest in minimizing the number of screenshots, perhaps just showing the modal is enough. Lmk if you have other thoughts

Image 2022-04-21 at 2 14 14 PM
Image 2022-04-21 at 2 14 27 PM
Image 2022-04-21 at 2 17 19 PM

**Comment by taroface** [comment]
@andf-crl This has some TKs as placeholders for some info we are awaiting from Datadog. Please disregard them for now. 

@abarganier Let me know when we have a name for the default dashboard! Thanks.

**Review by andf-crl — APPROVED** [review]
Nice work! Two questions for you, along with some small nits and suggestions. Approving for release on my end, but feel free to send it back my way if you want another copy round after tech additions are clarified from upstream. Thanks for your work on this!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13447#-:-N1Fn_pqDcdFeQ5_Zv0x:b-3xtqid)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier, @taroface, and @thtruo)

___
*[`cockroachcloud/monitoring-page.md` line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FfQTV01fSXOXFk8bp:-N1FfQTV01fSXOXFk8bq:b-5ijz37) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L13)):*
> ```Markdown
> - Access the cluster's [built-in DB Console](#access-the-db-console) to view time-series data on SQL queries, troubleshoot query performance, view a list of jobs, and more.
> 
> To view the **Monitoring** page, [log in](https://cockroachlabs.cloud/) and click **Monitoring**.
> ```

nit: can we provide any further guidance on where **Monitoring** can be found on-page? I do not see this anywhere on a free Serverless cluster's landing page on login -- and have not found out how to spin up a Dedicated Server for free yet -- so I can't offer any suggestions currently. I think this is worth clarifying because the rest of this document assumes that the user has successfully found the **Monitoring** page.
___
*[`cockroachcloud/monitoring-page.md` line 17 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FfxT-0SBelx-vqsIF:-N1FfxT-0SBelx-vqsIG:b-psebmq) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L17)):*
> ```Markdown
> ## Monitor with Datadog
> 
> The {{ site.data.products.dedicated }} integration for Datadog enables data collection and alerting on [CockroachDB metrics](https://docs.datadoghq.com/integrations/tk) using the Datadog platform.
> ```

Question: does DD support all CRDB metrics? If not, we can say something like [supported CockroachDB metrics] to cover this discrepancy.
___
*[`cockroachcloud/monitoring-page.md` line 31 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FgSUmCmr1H6Y1aTGt:-N1FgSUmCmr1H6Y1aTGu:bpp55it) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L31)):*
> ```Markdown
> 1. On the cluster's **Monitoring** page, click **Setup** in the **Datadog** panel.
> 
> 1. Fill in the **API key** and **Datadog Site** fields with the corresponding values.
> ```

nit: the corresponding --> their corresponding
___
*[`cockroachcloud/monitoring-page.md` line 33 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FghdVA2foui7igcGh:-N1FghdVA2foui7igcGi:b3mabzb) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L33)):*
> ```Markdown
> 1. Fill in the **API key** and **Datadog Site** fields with the corresponding values.
>     - The **API key** is associated with your Datadog organization. If you don't have an API key to use with your {{ site.data.products.dedicated }} cluster, you need to create one. For instructions, see the [Datadog documentation](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token).
>     - Your **Datadog Site** corresponds to your Datadog URL. For more details, see the [Datadog documentation](https://docs.datadoghq.com/getting_started/site/).
> ```

nit: Datadog appears to call this their "Site URL". "Datadog Site URL" is fine here.
___
*[`cockroachcloud/monitoring-page.md` line 35 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1Fh1EsBDjK5tqANzcN:-N1Fh1EsBDjK5tqANzcO:btt34li) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L35)):*
> ```Markdown
>     - Your **Datadog Site** corresponds to your Datadog URL. For more details, see the [Datadog documentation](https://docs.datadoghq.com/getting_started/site/).
> 
> 1. Click **Create**. The integration will take approximately 15 seconds to 2 minutes to enable, depending on the size of your cluster and the current load on the system.
> ```

nit: best to avoid prognostication unless we've testing a range of site size & load integrations. The following might set expectations without risking incorrectness for large sites:

"""
Depending on the size of your cluster and the current load on the system, the integration might take some time to become enabled.
"""
___
*[`cockroachcloud/monitoring-page.md` line 37 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FhV4a7lLy31wp6r-q:-N1FhV4a7lLy31wp6r-r:b-hqzjzl) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L37)):*
> ```Markdown
> 1. Click **Create**. The integration will take approximately 15 seconds to 2 minutes to enable, depending on the size of your cluster and the current load on the system.
> 
> 1. When enabled, the **Integration status** in the panel will show as `Active`.
> ```

nit: When --> Once
___
*[`cockroachcloud/monitoring-page.md` line 41 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1Fhs5t3Ue5-N5jgOxu:-N1Fhs5t3Ue5-N5jgOxv:b-r3mycu) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L41)):*
> ```Markdown
>     The cluster will also appear on your Datadog [Infrastructure List](https://docs.datadoghq.com/infrastructure/list/) once it is registered on Datadog. This can take up to several minutes.
> 
> ### Verify integration status
> ```

This section somewhat duplicates step 4 above. What if we removed step 4, and started `### Verify` section with something like:

"""
Once enabled, the **Integration status** in the **Datadog** panel will show as `Active`. 

If an issue is encountered during the integration, one of the following statuses may appear instead:

- `Inactive` ...
- `Unhealthy` ...
- `Unknown` ...
"""
___
*[`cockroachcloud/monitoring-page.md` line 48 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FjeXU1pKByRrdiqVN:-N1FjeXU1pKByRrdiqVO:bg2wdkx) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L48)):*
> ```Markdown
> - `Inactive` indicates that the integration has not been successfully deployed. Setup has either not been attempted or has encountered an error.
> - `Unhealthy` indicates that the integration API key is invalid and needs to be [updated](#update-integration).
> - `Unknown` indicates that an unknown error has occurred. If this status is displayed, contact our [support team](https://support.cockroachlabs.com/).
> ```

nit: expand link text to include verb: `[contact our support team]`

x2 on this page.
___
*[`cockroachcloud/monitoring-page.md` line 52 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FjXQI5nxsCO1BJoS0:-N1FjXQI5nxsCO1BJoS1:b815kvs) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L52)):*
> ```Markdown
> Metrics export from CockroachDB can be interrupted in the event of:
> 
> - A stale API key. In this case, the integration status will be `Unhealthy`. To resolve the issue, [update](#update-integration) your integration with a new API key.
> ```

nit: let's link the whole phrase: `update your integration`
___
*[`cockroachcloud/monitoring-page.md` line 53 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FjvuZCmUtZ-C5mNnF:-N1FjvuZCmUtZ-C5mNnG:b-aj9udm) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L53)):*
> ```Markdown
> 
> - A stale API key. In this case, the integration status will be `Unhealthy`. To resolve the issue, [update](#update-integration) your integration with a new API key.
> - Any issue that causes transient CockroachDB unavailbility. In this case, the integration status will continue to be `Active`. To resolve the issue, try [deactivating](#deactivate-integration) and reactivating the integration from the **Datadog** panel. If this does not resolve the issue, contact our [support team](https://support.cockroachlabs.com/).
> ```

nit: We can more closely match the structure of the first UL with the following change or similar to the first sentence:

"""
- Transient CockroachDB availability.
"""
___
*[`cockroachcloud/monitoring-page.md` line 53 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FkGNGBXPNtojxw6p2:-N1FkGNGBXPNtojxw6p3:bm6qw5t) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L53)):*
> ```Markdown
> 
> - A stale API key. In this case, the integration status will be `Unhealthy`. To resolve the issue, [update](#update-integration) your integration with a new API key.
> - Any issue that causes transient CockroachDB unavailbility. In this case, the integration status will continue to be `Active`. To resolve the issue, try [deactivating](#deactivate-integration) and reactivating the integration from the **Datadog** panel. If this does not resolve the issue, contact our [support team](https://support.cockroachlabs.com/).
> ```

Question: what are the user-facing effects of transient CRDB availability on the DD integration? Integration Status remaining `active` is important to know, but the user is likely to face something like: missing metric values over an arbitrary timespan, dropped graphs, holes in the data, etc. Is there a brief indication of this we can add to the `In this case` sentence to highlight what the user is likely to see? Maybe something like the following (total guess at the actual behavior!!):

"""
In this case, the integration status will continue to be `Active` but you might experience incomplete metrics exports in Datadog. 
"""
___
*[`cockroachcloud/monitoring-page.md` line 71 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FlMXF-y1OQ2SHFTth:-N1FlMXF-y1OQ2SHFTti:b-p9gsec) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L71)):*
> ```Markdown
> To preview the metrics being collected, you can:
> 
> - Click on your cluster's entry in the [Infrastructure List](https://docs.datadoghq.com/infrastructure/list/). The time-series graphs for each available metric are displayed.
> ```

nit: you could safely combine these sentences, perhaps similar to:

"""
- Click on your cluster's entry in the [Infrastructure List](https://docs.datadoghq.com/infrastructure/list/). to display time-series graphs for each available metric.
"""
___
*[`cockroachcloud/monitoring-page.md` line 76 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1Flf7_6F6b8wL22H2Q:-N1Flf7_6F6b8wL22H2R:b1yam0o) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L76)):*
> ```Markdown
> ### Monitor health of metrics export
> 
> To monitor the health of metrics export, we recommend that you create a new Monitor, following the steps in the [Datadog documentation](https://docs.datadoghq.com/monitors/create/types/metric/?tab=threshold).
> ```

We should link the task directly, similarly to how we do this elsewhere in this doc:

"""
To monitor the health of metrics export, we recommend that you [create a new Monitor](https://docs.datadoghq.com/monitors/create/types/metric/?tab=threshold)."""
___
*[`cockroachcloud/monitoring-page.md` line 78 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1Fm5ED7iMiMlw-3gP1:-N1Fm5ED7iMiMlw-3gP2:b-gy8033) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L78)):*
> ```Markdown
> To monitor the health of metrics export, we recommend that you create a new Monitor, following the steps in the [Datadog documentation](https://docs.datadoghq.com/monitors/create/types/metric/?tab=threshold).
> 
> Select **Threshold Alert** as the detection method. You can use this option to configure an alert that is sent when a [supported metric](https://docs.datadoghq.com/integrations/cockroachdb/?tab=host#data-collected) reaches a given threshold. For descriptions of some useful CockroachDB alerts, see [Monitoring and Alerting](../{{site.versions["stable"]}}/monitoring-and-alerting.html#events-to-alert-on).
> ```

nit: we can combine these first two sentences I think. Something like:

"""
Select **Threshold Alert** as the detection method, which configures an alert that is sent when a [supported metric](https://docs.datadoghq.com/integrations/cockroachdb/?tab=host#data-collected) reaches a given threshold.
"""

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13447)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier, @andf-crl, @taroface, and @thtruo)

___
*[`cockroachcloud/monitoring-page.md` line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FfQTV01fSXOXFk8bp:-N1KWj1WDS6uWY2lpEqb:b-m0hf9z) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L13)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: can we provide any further guidance on where **Monitoring** can be found on-page? I do not see this anywhere on a free Serverless cluster's landing page on login -- and have not found out how to spin up a Dedicated Server for free yet -- so I can't offer any suggestions currently. I think this is worth clarifying because the rest of this document assumes that the user has successfully found the **Monitoring** page.
</blockquote></details>

Done. Also, one of the Obs teams should be able to set you up with access to view/create staging clusters!
___
*[`cockroachcloud/monitoring-page.md` line 17 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FfxT-0SBelx-vqsIF:-N1KX8YX4kWh2gPw8ytG:bsniewf) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L17)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

Question: does DD support all CRDB metrics? If not, we can say something like [supported CockroachDB metrics] to cover this discrepancy.
</blockquote></details>

I believe so! Amended to reflect this. @thtruo @abarganier please correct if this is wrong.
___
*[`cockroachcloud/monitoring-page.md` line 33 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FghdVA2foui7igcGh:-N1KXHhw8cQI0k0IjHTt:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L33)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: Datadog appears to call this their "Site URL". "Datadog Site URL" is fine here.
</blockquote></details>

Done.
___
*[`cockroachcloud/monitoring-page.md` line 35 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1Fh1EsBDjK5tqANzcN:-N1KXLLJEfDK4NyzVnJU:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L35)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: best to avoid prognostication unless we've testing a range of site size & load integrations. The following might set expectations without risking incorrectness for large sites:

"""
Depending on the size of your cluster and the current load on the system, the integration might take some time to become enabled.
"""
</blockquote></details>

Done.
___
*[`cockroachcloud/monitoring-page.md` line 37 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FhV4a7lLy31wp6r-q:-N1KXOYTECCESrsOjJnt:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L37)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: When --> Once
</blockquote></details>

Done.
___
*[`cockroachcloud/monitoring-page.md` line 41 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1Fhs5t3Ue5-N5jgOxu:-N1KXlON2gTacGwqcWfm:b-896fja) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L41)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

This section somewhat duplicates step 4 above. What if we removed step 4, and started `### Verify` section with something like:

"""
Once enabled, the **Integration status** in the **Datadog** panel will show as `Active`. 

If an issue is encountered during the integration, one of the following statuses may appear instead:

- `Inactive` ...
- `Unhealthy` ...
- `Unknown` ...
"""
</blockquote></details>

Done!
___
*[`cockroachcloud/monitoring-page.md` line 48 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FjeXU1pKByRrdiqVN:-N1KXrtQ-6T56VkirvwB:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L48)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: expand link text to include verb: `[contact our support team]`

x2 on this page.
</blockquote></details>

Done.
___
*[`cockroachcloud/monitoring-page.md` line 52 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FjXQI5nxsCO1BJoS0:-N1KXyKp80HUqjpBGnQe:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L52)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: let's link the whole phrase: `update your integration`
</blockquote></details>

Done.
___
*[`cockroachcloud/monitoring-page.md` line 53 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FjvuZCmUtZ-C5mNnF:-N1KYJO1DmdZgUdcxrmI:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L53)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: We can more closely match the structure of the first UL with the following change or similar to the first sentence:

"""
- Transient CockroachDB availability.
"""
</blockquote></details>

Done.
___
*[`cockroachcloud/monitoring-page.md` line 53 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FkGNGBXPNtojxw6p2:-N1KYO3B84KYKkEj9eWI:b-hcihae) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L53)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

Question: what are the user-facing effects of transient CRDB availability on the DD integration? Integration Status remaining `active` is important to know, but the user is likely to face something like: missing metric values over an arbitrary timespan, dropped graphs, holes in the data, etc. Is there a brief indication of this we can add to the `In this case` sentence to highlight what the user is likely to see? Maybe something like the following (total guess at the actual behavior!!):

"""
In this case, the integration status will continue to be `Active` but you might experience incomplete metrics exports in Datadog. 
"""
</blockquote></details>

I agree this would be good to clarify. @abarganier could you provide a suggestion / comment on how the above sounds to you?
___
*[`cockroachcloud/monitoring-page.md` line 71 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FlMXF-y1OQ2SHFTth:-N1KYbtO1emmx3XU76rf:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L71)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: you could safely combine these sentences, perhaps similar to:

"""
- Click on your cluster's entry in the [Infrastructure List](https://docs.datadoghq.com/infrastructure/list/). to display time-series graphs for each available metric.
"""
</blockquote></details>

Done.
___
*[`cockroachcloud/monitoring-page.md` line 76 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1Flf7_6F6b8wL22H2Q:-N1KYtI965Ot8UeleW-i:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L76)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

We should link the task directly, similarly to how we do this elsewhere in this doc:

"""
To monitor the health of metrics export, we recommend that you [create a new Monitor](https://docs.datadoghq.com/monitors/create/types/metric/?tab=threshold)."""
</blockquote></details>

Done.
___
*[`cockroachcloud/monitoring-page.md` line 78 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1Fm5ED7iMiMlw-3gP1:-N1KYwe82UdDXKzwJPjB:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L78)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: we can combine these first two sentences I think. Something like:

"""
Select **Threshold Alert** as the detection method, which configures an alert that is sent when a [supported metric](https://docs.datadoghq.com/integrations/cockroachdb/?tab=host#data-collected) reaches a given threshold.
"""

</blockquote></details>

Done.

**Review by thtruo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13447)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier, @andf-crl, @taroface, and @thtruo)

___
*[`cockroachcloud/monitoring-page.md` line 17 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FfxT-0SBelx-vqsIF:-N1QEuMCBfttzQqrRhfJ:b-a6tep5) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L17)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

I believe so! Amended to reflect this. @thtruo @abarganier please correct if this is wrong.
</blockquote></details>

Note that the DD integration only supports a subset of CDRDB metrics (specifically only 237 of them). If you want to call out that nuance, we might want to consider adding language to reflect that it's a subset and/or point readers back to DDOG's documentation which will include a glossary of those CRDB metrics.

**Review by abarganier — COMMENTED** [review]
> @abarganier Let me know when we have a name for the default dashboard! Thanks.

Will do! As soon as it's registered on their end I will let you know. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13447#-:-N1dwHgdALJqPHUDmhyp:byu127c)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier, @andf-crl, @taroface, and @thtruo)

___
*[`cockroachcloud/monitoring-page.md` line 46 at r2](https://reviewable.io/reviews/cockroachdb/docs/13447#-MzWjoo6IODTZ4AwvVf8-r2-46:-N1dwOqO27JJfISQQ6gZ:bxzk581) ([raw file](https://github.com/cockroachdb/docs/blob/9b296b6beb499df4aaaeed0bd0ac648d2c7c7f11/cockroachcloud/monitoring-page.md#L46)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Good point - how about:

"Setup has either not been attempted or has encountered an error."
</blockquote></details>

Sorry for missing this - LGTM!
___
*[`cockroachcloud/monitoring-page.md` line 17 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FfxT-0SBelx-vqsIF:-N1duEbw5mnoLnGLRqRL:b9jvjpl) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L17)):*
<details><summary><i>Previously, thtruo (Tommy) wrote…</i></summary><blockquote>

Note that the DD integration only supports a subset of CDRDB metrics (specifically only 237 of them). If you want to call out that nuance, we might want to consider adding language to reflect that it's a subset and/or point readers back to DDOG's documentation which will include a glossary of those CRDB metrics. 
</blockquote></details>

+1 for calling out that the metric set is a subset - I imagine that would be a common inquiry by customers.
___
*[`cockroachcloud/monitoring-page.md` line 53 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FkGNGBXPNtojxw6p2:-N1dv-G3-N9w_bgQNi7_:b-mazgrn) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L53)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

I agree this would be good to clarify. @abarganier could you provide a suggestion / comment on how the above sounds to you?
</blockquote></details>

I think that sounds good - you're right that the end result for the customer would be gaps in the timeseries where there are no metric values. It might not be worth outlining in the docs, but aside from API key issues or a Datadog server outage, this could happen for 2 reasons:

1. The OpenTelemetry Collector is having trouble, so it's not scraping + forwarding metrics.
2. The CockroachDB node is unavailable/not serving metrics, so the otel collector has nothing to scrape + forward.

It might be worth noting that gaps in metrics within Datadog do not necessarily point to an availability issue with the CockroachDB node or cluster itself (such as in 1 above), but that it's worth reaching out to support regardless since it's something our engineers should take a look at.

**Comment by thtruo** [comment]
Update: it seems like we will have the billing exception in place (concrete date tbd), so never mind my previous comment about adding a disclaimer.

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13447)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier, @andf-crl, @taroface, and @thtruo)

___
*[`cockroachcloud/monitoring-page.md` line 17 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FfxT-0SBelx-vqsIF:-N2DRj53AkVq_n5U2yv5:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L17)):*
<details><summary><i>Previously, abarganier (Alex Barganier) wrote…</i></summary><blockquote>

+1 for calling out that the metric set is a subset - I imagine that would be a common inquiry by customers.
</blockquote></details>

Done.
___
*[`cockroachcloud/monitoring-page.md` line 53 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FkGNGBXPNtojxw6p2:-N2DU1SACSQGKuDNWNZp:bw3bf1m) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L53)):*
<details><summary><i>Previously, abarganier (Alex Barganier) wrote…</i></summary><blockquote>

I think that sounds good - you're right that the end result for the customer would be gaps in the timeseries where there are no metric values. It might not be worth outlining in the docs, but aside from API key issues or a Datadog server outage, this could happen for 2 reasons:

1. The OpenTelemetry Collector is having trouble, so it's not scraping + forwarding metrics.
2. The CockroachDB node is unavailable/not serving metrics, so the otel collector has nothing to scrape + forward.

It might be worth noting that gaps in metrics within Datadog do not necessarily point to an availability issue with the CockroachDB node or cluster itself (such as in 1 above), but that it's worth reaching out to support regardless since it's something our engineers should take a look at.
</blockquote></details>

Noted. I added the OpenTelemetry Collector to the bullet about CRDB unavailability. I think this should convey all points!

**Review by abarganier — COMMENTED** [review]
@taroface we have our placeholder URL for Datadog's docs! Here you go: https://docs.datadoghq.com/integrations/cockroach_cloud/

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13447#-:-N2NANyl8wSRm2pqNBWt:byefsg5)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier, @andf-crl, @taroface, and @thtruo)

___
*[`cockroachcloud/monitoring-page.md` line 53 at r5](https://reviewable.io/reviews/cockroachdb/docs/13447#-N1FkGNGBXPNtojxw6p2:-N2N7b122xORYlGBc3QU:bx1udmd) ([raw file](https://github.com/cockroachdb/docs/blob/76b030843296ed5415d399e11067e44abcc08639/cockroachcloud/monitoring-page.md#L53)):*
> Transient CockroachDB unavailbility or issues with the OpenTelemetry Collector that exports the metrics. In each case, the integration status will continue to be `Active`

I don't think this detail about the `Active` status is necessarily accurate. When we determine the status of the collector, we check to see that the collector is in a "ready" state (e.g. all the processes/pods we expect to exist are up and running). If that's not the case, an error status will show. 

It's possible that there's some problem with the collector that doesn't cause it to crash, in which case it would indeed still show an `Active` status... But maybe we should just keep things simple and say that the status will be `Error` if there's a problem with the collector :shrug:. I think the primary thing that comes to mind in terms of things that wouldn't crash the collector is the stale API key issue, and we now have a separate status for that :slightly_smiling_face: 

Apologies if my last message was confusing on this!

**Comment by abarganier** [comment]
@taroface sorry for the churn, the URL is actually:  https://docs.datadoghq.com/integrations/cockroach-cloud

**Comment by taroface** [comment]
@abarganier Thanks for this clarification. I may have missed this update earlier. So in addition to the four statuses documented here, there is a fifth `Error` status that encompasses issues with the OpenTelemetry Collector?

**Review by abarganier — COMMENTED** [review]
> @abarganier Thanks for this clarification. I may have missed this update earlier. So in addition to the four statuses documented here, there is a fifth Error status that encompasses issues (those encountered during the readiness check) with the OpenTelemetry Collector?

I just tested this scenario and it looks like the `Datadog integration rollout failed. Retry rollout` state shows with an `Inactive` status if the collector becomes unavailable. We should probably have some other status for this 🤔 but since it's an unlikely occurrence, and the customer has a path forward (retrying the rollout), it's unlikely it will be done before we launch into GA. 



> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13447#-:-N2S9-jA73qZ94PDK5JN:b9eqp5p)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier, @andf-crl, @taroface, and @thtruo)

**Pull request merged**
